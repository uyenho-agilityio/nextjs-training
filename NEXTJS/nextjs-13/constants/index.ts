export const LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const PAGE_404 = {
  src: '/images/404.jpg',
  alt: '404 Page',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const ROUTE_LIST = [
  {
    label: 'Home',
    path: '/',
  },
  {
    label: 'Events',
    path: '/events',
  },
  {
    label: 'About',
    path: '/about',
  },
];

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};

export const SEO_DATA = {
  title: 'NextJS Training with app/ directory',
  description: 'Do examples to understand NextJS 13',
  icon: '/favicon.ico',
  url: 'https://nextjs-examples-training.vercel.app/',
  thumbnailUrl:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8e/Nextjs-logo.svg/1200px-Nextjs-logo.svg.png',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
export const LOCAL_BASE_URL = process.env.NEXT_PUBLIC_LOCAL_BASE_URL;
