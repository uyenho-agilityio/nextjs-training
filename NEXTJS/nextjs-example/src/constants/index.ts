export const VERCEL_LOGO = {
  alt: 'Vercel Logo',
  src: '/images/vercel.svg',
  width: 72,
  height: 16,
};

export const COURSE_LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const AVATAR = {
  alt: 'Avatar',
  src: '/images/avatar.jpg',
  width: 40,
  height: 40,
};

export const PAGE_404 = {
  alt: '404 Page',
  src: '/images/404.jpg',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_TO_COURSE: 'Back to Course',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};
