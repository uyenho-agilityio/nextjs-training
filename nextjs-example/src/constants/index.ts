export const VERCEL_LOGO = {
  alt: 'Vercel Logo',
  src: '/images/vercel.svg',
  width: 72,
  height: 16,
};

export const COURSE_LOGO = {
  alt: 'Logo',
  src: '/images/logo.svg',
  width: 32,
  height: 32,
};

export const AVATAR = {
  alt: 'Avatar',
  src: '/images/avatar.jpg',
  width: 40,
  height: 40,
};

export const PAGE_404 = {
  alt: '404 Page',
  src: '/images/404.jpg',
  width: 400,
  height: 400,
};

export const PAGE_ROUTES = {
  HOME: '/',
  ABOUT: '/about',
  EVENTS: '/events',
};

export const ROUTE_LIST = [
  {
    label: 'Home',
    path: '/',
  },
  {
    label: 'Events',
    path: '/events',
  },
  {
    label: 'About',
    path: '/about',
  },
  {
    label: 'NextJS',
    path: '/nextjs',
  },
];

export const MESSAGES = {
  NETWORK_ERROR: 'Network error. Please try again later.',
  ERROR: 'Something went wrong!',
};

export const TEXT = {
  CONTINUE: 'Continue',
  BACK_TO_COURSE: 'Back to Course',
  BACK_HOME: 'BACK TO HOME',
  HEAD_BACK: "Let's Head Back!",
  SIGN_IN: 'Sign in',
  ACADEMY: 'Academy',
};

export const SEO_DATA = {
  title: 'NextJS Training',
  description: 'Do examples to understand NextJS',
  icon: '/images/favicon.ico',
  url: 'https://nextjs-examples-training.vercel.app/',
  thumbnailUrl:
    'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8e/Nextjs-logo.svg/1200px-Nextjs-logo.svg.png',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
