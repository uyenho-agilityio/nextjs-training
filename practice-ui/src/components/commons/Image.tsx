import React from 'react';
import { ImgVariants } from '~enums/variant';

export interface ImgProps {
  imageUrl: string;
  className?: string;
  alt: string;
  height?: string;
  width?: string;
  variant?: ImgVariants;
}

export const Image: React.FC<ImgProps> = React.memo(
  ({ imageUrl, className = '', alt, height, width, variant = ImgVariants.Image }): JSX.Element => {
    return (
      <img
        srcSet={imageUrl}
        className={`${variant === ImgVariants.Icon ? 'navbar-toggler-icon' : `${className}`}`}
        alt={alt}
        style={{ width: width, height: height }}
      />
    );
  },
);
