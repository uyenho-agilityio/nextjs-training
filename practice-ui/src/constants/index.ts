export * from './footer';
export * from './header';
export * from './management';
export * from './pricing';
export * from './product';
