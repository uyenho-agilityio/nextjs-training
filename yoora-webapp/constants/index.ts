export * from './common';
export * from './footer';
export * from './images';
export * from './routes';
export * from './sections';
