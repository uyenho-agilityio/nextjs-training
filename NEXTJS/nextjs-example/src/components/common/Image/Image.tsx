// Libs
import { Box, BoxProps, chakra } from '@chakra-ui/react';
import NextImage, { ImageProps } from 'next/image';
import React from 'react';

// Utils
import { generateBlurDataUrl } from 'src/utils';

// const loaderImage = (resolverProps: ImageLoaderProps): string => {
//   return `${resolverProps.src}?w=${resolverProps.width}&q=${resolverProps.quality ?? 80}`;
// };

const ChakraNextImage = chakra(NextImage, {
  shouldForwardProp: prop =>
    ['width', 'height', 'src', 'alt', 'placeholder', 'blurDataURL'].includes(prop),
});

export const Image: React.FC<ImageProps & BoxProps> = (props): JSX.Element => {
  const { src, alt, width = 1, height = 1, ...rest } = props;

  return (
    <Box pos="relative" cursor="pointer" className="group" {...rest}>
      {+width > 40 && +height > 40 ? (
        <ChakraNextImage
          // loader={loaderImage}
          width={width}
          height={height}
          src={src}
          alt={alt}
          placeholder="blur"
          blurDataURL={generateBlurDataUrl(+`${width}`, +`${height}`)}
        />
      ) : (
        <NextImage width={width} height={height} src={src} alt={alt} />
      )}
    </Box>
  );
};
