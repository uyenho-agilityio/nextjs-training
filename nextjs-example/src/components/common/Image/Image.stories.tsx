// Lib
import Storybook from '@storybook/react';

// Component
import { Image } from './Image';

// Constant
import { PAGE_404 } from 'src/constants';

export default {
  title: 'Components/Image',
  component: Image,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Storybook.ComponentMeta<typeof Image>;

const Template: Storybook.ComponentStory<typeof Image> = args => <Image {...args} />;

export const NotFound = Template.bind({});
NotFound.args = { ...PAGE_404, src: 'https://i.ibb.co/Mn3JMQc/404.jpg' };
