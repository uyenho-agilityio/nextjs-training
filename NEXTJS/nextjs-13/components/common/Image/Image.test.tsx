// Lib
import { render, screen, fireEvent } from '@testing-library/react';

// Component
import { Image } from './Image';

// Constant
import { LOGO } from 'constants/index';

describe('Image render', () => {
  test('should render image component with default props', () => {
    const { container } = render(<Image {...LOGO} />);
    expect(container).toMatchSnapshot();
  });

  test('should simulate click event and expect mock function to be called', () => {
    const mockOnClick = jest.fn();
    render(<Image onClick={mockOnClick} {...LOGO} />);

    const button = screen.getByAltText(/logo/i);
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
