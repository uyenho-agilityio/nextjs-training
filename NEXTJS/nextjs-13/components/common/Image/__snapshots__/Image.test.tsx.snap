// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image render should render image component with default props 1`] = `
<div>
  <div
    class="group css-dfpqc0"
  >
    <img
      alt="Logo"
      data-nimg="1"
      decoding="async"
      height="32"
      src="/images/logo.svg"
      srcset="/images/logo.svg 1x, /images/logo.svg 2x"
      style="color: transparent;"
      width="32"
    />
  </div>
</div>
`;
